airbyte-webapp/src/components/connectorBuilder/StreamTestingPanel/StreamTester.tsx

```diff
@@ -44,6 +52,25 @@ export const StreamTester: React.FC<{
     }
   }, [isError]);
 
+  useEffect(() => {
+    // This will only be true if the data was manually refetched by the user clicking the Test button,
+    // so the analytics events won't fire just from the user switching between streams, as desired
+    if (isFetchedAfterMount) {
+      if (errorMessage) {
+        analyticsService.track(Namespace.CONNECTOR_BUILDER, Action.STREAM_TEST_FAILURE, {
```

Comment:
This is called on every state update:
* Make an actual stream test which results in failure: works as expected
* Afterwards, change something in the UI - another failure event is recorded (even if the user didn't click on test).

This happens because `jsonManifest.streams` changes its reference on every state update.

Idea to fix this: Use the streams via ref and re-run the hook on `streamReadData`

Reply:
@flash1293 since we separately decided to not send the stream JSON body in these events, I no longer had the issue of this depending on `jsonManifest.streams` and re-firing for every state change. However, I did need to add a couple more changes here to have this `useEffect` be dependent on `dataUpdatedAt` and `errorUpdatedAt` from the react-query, to make the events be tracked when a user clicks Test again and the results don't change. See commit: https://github.com/airbytehq/airbyte/pull/21686/commits/09967f01a770cde0321f2f855e58f1cac96d8434

With these changes, I believe this is now working properly, but definitely lmk if you think anything else should be changed here.